lib/puppet/network/http/pool.rb

```diff
@@ -71,7 +71,8 @@ def close_connection(site, http)
   # @api private
   def borrow(site, verifier)
     @pool[site] = active_sessions(site)
-    session = @pool[site].shift
+    index = @pool[site].index { |session| verifier.reusable?(session.verifier) }
```

Comment:
To make sure I'm understanding this, this will return the index of the first session it finds that is reusable?

Reply:
Yeah, AFAIK ruby doesn't have an enumerable method to "delete the first element for which the block is true". The `Enumerable#delete_if` method unfortunately deletes **all** elements for which the block is true.